### PR TITLE
Fixed phpdoc

### PR DESCRIPTION
**Pull request by core23**

### CODE REVIEW

File: Admin/BaseMediaAdmin.php [hunk 1]
```diff
@@ -169,7 +175,7 @@ public function getNewInstance()
     }
 
     /**
-     * @return null|\Sonata\MediaBundle\Provider\Pool
+     * @return Pool|null
```
Comment (OskarStark) [inline Admin/BaseMediaAdmin.php]: this is wrong IMO, this can't be null, because we have typehint in the ``__construct()`` method and this comes from the DI

File: Admin/Manager/DoctrineMongoDBManager.php [hunk 2]
```diff
@@ -11,15 +13,18 @@
...
     protected $manager;
 
     /**
-     * @param mixed $entityManager
-     * @param mixed $manager
+     * @param ManagerRegistry  $registry
+     * @param ManagerInterface $manager
      */
-    public function __construct($entityManager, $manager)
+    public function __construct($registry, $manager)
```
Comment (OskarStark) [inline Admin/Manager/DoctrineMongoDBManager.php]: would name this ``$managerRegistry``instead of ``$registry``

File: CDN/CloudFront.php [hunk 3]
```diff
@@ -162,7 +162,7 @@ private function getClient()
     /**
      * For testing only.
      *
-     * @param $client
+     * @param CloudFrontClient $client
```
Comment (OskarStark) [inline CDN/CloudFront.php]: hmm, is this correct? then we need a type hint on ``setClient()``, don't we?

File: Command/AddMediaCommand.php [hunk 4]
```diff
@@ -18,6 +18,9 @@
 
 class AddMediaCommand extends BaseCommand
 {
+    /**
+     * @var bool
+     */
     protected $quiet = false;
     protected $output;
```
Comment (OskarStark) [inline Command/AddMediaCommand.php]: php doc for ``$output``missing

File: Controller/Api/MediaController.php [hunk 5]
```diff
@@ -374,7 +374,7 @@ protected function getMedium($id = null)
      * @param MediaInterface         $medium
      * @param MediaProviderInterface $provider
      *
-     * @return \FOS\RestBundle\View\View|\Symfony\Component\Form\Form
+     * @return View|FormInterface
      */
     protected function handleWriteMedium(Request $request, MediaInterface $medium, MediaProviderInterface $provider)
```
Comment (OskarStark) [inline Controller/Api/MediaController.php]: what about renaming ``$medium``to ``$media``, this looks like a typo to me

File: Extra/Pixlr.php [hunk 6]
```diff
@@ -24,32 +24,59 @@
...
+     */
     protected $container;
 
+    /**
+     * @var string[]
+     */
     protected $validFormats;
 
+    /**
+     * @var string
```
Comment (OskarStark) [inline Extra/Pixlr.php]: sounds like a bool, but didn't verify this

File: Extra/Pixlr.php [hunk 7]
```diff
@@ -24,32 +24,59 @@
...
-     * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
+     * @param string                $referrer
+     * @param string                $secret
+     * @param Pool                  $pool
+     * @param MediaManagerInterface $mediaManager
+     * @param RouterInterface       $router
+     * @param EngineInterface       $templating
+     * @param ContainerInterface    $container
      */
     public function __construct($referrer, $secret, Pool $pool, MediaManagerInterface $mediaManager, RouterInterface $router, EngineInterface $templating, ContainerInterface $container)
```
Comment (OskarStark) [inline Extra/Pixlr.php]: we should add line breaks here, for better readability

File: Generator/GeneratorInterface.php [hunk 8]
```diff
@@ -16,9 +16,7 @@
 interface GeneratorInterface
 {
     /**
-     * @abstract
```
Comment (OskarStark) [inline Generator/GeneratorInterface.php]: why removing this? @core23 

ping @Soullivaneuh

File: Generator/ODMGenerator.php [hunk 9]
```diff
@@ -16,11 +16,7 @@
 class ODMGenerator implements GeneratorInterface
 {
     /**
-     * @abstract
```
Comment (OskarStark) [inline Generator/ODMGenerator.php]: same here?

File: Listener/BaseMediaEventSubscriber.php [hunk 10]
```diff
@@ -38,25 +41,21 @@ public function getPool()
     }
 
     /**
-     * @abstract
```
Comment (OskarStark) [inline Listener/BaseMediaEventSubscriber.php]: same here, please check the other ``@abstract``-phpdocs's

File: Listener/ORM/MediaEventSubscriber.php [hunk 11]
```diff
@@ -69,7 +73,7 @@ protected function getMedia(EventArgs $args)
     /**
      * @param MediaInterface $media
      *
-     * @return mixed
+     * @return CategoryInterface[]
```
Comment (OskarStark) [inline Listener/ORM/MediaEventSubscriber.php]: this returns only one CategoryInterface IMO not an array

Comment (OskarStark) [inline Listener/ORM/MediaEventSubscriber.php]: and this mehtods throws an exception

File: Metadata/ProxyMetadataBuilder.php [hunk 12]
```diff
@@ -17,12 +17,20 @@
...
+     * @var ContainerInterface
+     */
     private $container;
+
+    /**
+     * @var array
+     */
     private $map;
+
     private $metadata;
```
Comment (OskarStark) [inline Metadata/ProxyMetadataBuilder.php]: missing phpdoc

File: Metadata/ProxyMetadataBuilder.php [hunk 13]
```diff
@@ -17,12 +17,20 @@
 
 class ProxyMetadataBuilder implements MetadataBuilderInterface
 {
+    /**
+     * @var ContainerInterface
+     */
     private $container;
+
+    /**
+     * @var array
```
Comment (OskarStark) [inline Metadata/ProxyMetadataBuilder.php]: not ``string[]`` ?

File: Metadata/ProxyMetadataBuilder.php [hunk 14]
```diff
@@ -53,7 +61,7 @@ public function get(MediaInterface $media, $filename)
     }
 
     /**
-     * @param MediaInterface $metadata
+     * @param MediaInterface $media
      * @param string         $filename
      *
      * @return array
```
Comment (OskarStark) [inline Metadata/ProxyMetadataBuilder.php]: could also return ``false``

File: PHPCR/MediaManager.php [hunk 15]
```diff
@@ -17,10 +17,6 @@ class MediaManager extends BasePHPCRManager
 {
     /**
      * {@inheritdoc}
-     *
-     * Warning: previous method signature was : save(MediaInterface $media, $context = null, $providerName = null)
```
Comment (OskarStark) [inline PHPCR/MediaManager.php]: not sure about removing this

ping @Soullivaneuh

File: Provider/BaseVideoProvider.php [hunk 16]
```diff
@@ -26,16 +26,24 @@
...
-     * @param \Buzz\Browser                                    $browser
+     * @param string                        $name
+     * @param Filesystem                    $filesystem
+     * @param CDNInterface                  $cdn
+     * @param GeneratorInterface            $pathGenerator
+     * @param ThumbnailInterface            $thumbnail
+     * @param Browser                       $browser
+     * @param MetadataBuilderInterface|null $metadata
      */
     public function __construct($name, Filesystem $filesystem, CDNInterface $cdn, GeneratorInterface $pathGenerator, ThumbnailInterface $thumbnail, Browser $browser, MetadataBuilderInterface $metadata = null)
```
Comment (OskarStark) [inline Provider/BaseVideoProvider.php]: would add line breaks here for better readability

File: Provider/BaseVideoProvider.php [hunk 17]
```diff
@@ -194,11 +202,11 @@ protected function getMetadata(MediaInterface $media, $url)
...
-     * @param string                                   $format
-     * @param array                                    $options
+     * @param MediaInterface $media
+     * @param string         $format
+     * @param array          $options
      *
-     * @return \Imagine\Image\Box
+     * @return Box
      */
     protected function getBoxHelperProperties(MediaInterface $media, $format, $options = array())
```
Comment (OskarStark) [inline Provider/BaseVideoProvider.php]: this should be a separate PR, but we should add an ``array``-typehint to ``$options`` IMO

File: Provider/FileProvider.php [hunk 18]
```diff
@@ -40,14 +40,14 @@ class FileProvider extends BaseProvider
...
+     * @param string                   $name
+     * @param Filesystem               $filesystem
+     * @param CDNInterface             $cdn
+     * @param GeneratorInterface       $pathGenerator
+     * @param ThumbnailInterface       $thumbnail
+     * @param array                    $allowedExtensions
+     * @param array                    $allowedMimeTypes
+     * @param MetadataBuilderInterface $metadata
      */
     public function __construct($name, Filesystem $filesystem, CDNInterface $cdn, GeneratorInterface $pathGenerator, ThumbnailInterface $thumbnail, array $allowedExtensions = array(), array $allowedMimeTypes = array(), MetadataBuilderInterface $metadata = null)
```
Comment (OskarStark) [inline Provider/FileProvider.php]: i would add line breaks for better readability, this could be applied in a separate PR, for the other long constructor methods, too.

File: Provider/Pool.php [hunk 19]
```diff
@@ -22,10 +22,19 @@ class Pool
      */
     protected $providers = array();
 
+    /**
+     * @var array
```
Comment (OskarStark) [inline Provider/Pool.php]: ``string[]`` ?
not sure, can you verify this @core23 ? Thank you

File: Provider/Pool.php [hunk 20]
```diff
@@ -68,8 +77,8 @@ public function addProvider($name, MediaProviderInterface $instance)
     }
 
     /**
-     * @param string                                                 $name
-     * @param \Sonata\MediaBundle\Security\DownloadStrategyInterface $security
+     * @param string                    $name
+     * @param DownloadStrategyInterface $security
      */
     public function addDownloadSecurity($name, DownloadStrategyInterface $security)
```
Comment (OskarStark) [inline Provider/Pool.php]: i think we should rename ``$security`` to ``$strategy`` in a separate PR

and we then need BC handling for ``protected $downloadSecurities``

File: Twig/Extension/MediaExtension.php [hunk 21]
```diff
@@ -20,12 +20,24 @@
...
+     */
     protected $resources = array();
 
+    /**
+     * @var ManagerInterface
+     */
     protected $mediaManager;
 
+    /**
+     * @var
```
Comment (OskarStark) [inline Twig/Extension/MediaExtension.php]: ??

File: Twig/Extension/MediaExtension.php [hunk 22]
```diff
@@ -99,7 +111,7 @@ public function media($media = null, $format, $options = array())
     /**
      * @param mixed $media
      *
-     * @return null|\Sonata\MediaBundle\Model\MediaInterface
+     * @return MediaInterface|null
```
Comment (OskarStark) [inline Twig/Extension/MediaExtension.php]: could also return false

**Comment by OskarStark** [comment]
Great work @core23 , can you please apply my comments and rebase this PR?

Thank you :+1:

File: Extra/Pixlr.php (hunk 6 above)
Comment (core23) [inline Extra/Pixlr.php]: ... but is a string ;)
https://github.com/sonata-project/SonataMediaBundle/blob/master/Extra/Pixlr.php#L65

File: Generator/GeneratorInterface.php (hunk 8 above)
Comment (core23) [inline Generator/GeneratorInterface.php]: As far as I know, they are needless because the function is already abstract.

File: Metadata/ProxyMetadataBuilder.php (hunk 12 above)
Comment (core23) [inline Metadata/ProxyMetadataBuilder.php]: This property isn't used anywhere. I'll create another PR for this.

File: Metadata/ProxyMetadataBuilder.php (hunk 13 above)
Comment (core23) [inline Metadata/ProxyMetadataBuilder.php]: Same here. Never used

File: Provider/Pool.php (hunk 19 above)
Comment (core23) [inline Provider/Pool.php]: No it's an array of arrays

File: Extra/Pixlr.php (hunk 7 above)
Comment (core23) [inline Extra/Pixlr.php]: Where do you want breaks?

File: CDN/CloudFront.php (hunk 3 above)
Comment (core23) [inline CDN/CloudFront.php]: Should be correct, as the CloudFrontClient::factory method returns a CloudFrontClient.
I'll create a seperate PR for the type check.

File: Extra/Pixlr.php (hunk 7 above)
Comment (Soullivaneuh) [inline Extra/Pixlr.php]: @OskarStark This line is OK IMO.

File: Generator/GeneratorInterface.php (hunk 8 above)
Comment (Soullivaneuh) [inline Generator/GeneratorInterface.php]: Hmm not sure about that, the method is not declared as abstract.

But I'm wondering the utility of this tag... cc @rande.

File: PHPCR/MediaManager.php (hunk 15 above)
Comment (Soullivaneuh) [inline PHPCR/MediaManager.php]: Wow, thas was here a long time before I contribute to sonata-project. :-)

ping @rande instead.

**Comment by OskarStark** [comment]
Thank you @core23

File: PHPCR/MediaManager.php (hunk 15 above)
Comment (rande) [inline PHPCR/MediaManager.php]: 2011 ;) https://github.com/sonata-project/SonataMediaBundle/commit/484aec5a6b58924639b15de26c0cb7657c45ed81
